pkg/kubelet/status/manager.go

```diff
@@ -182,16 +195,42 @@ func (m *manager) TerminatePods(pods []*api.Pod) bool {
 				Terminated: &api.ContainerStateTerminated{},
 			}
 		}
-		select {
-		case m.podStatusChannel <- podStatusSyncRequest{pod, pod.Status}:
-		default:
+		newStatus := m.updateStatusInternal(pod, pod.Status)
+		if newStatus != nil {
+			select {
+			case m.podStatusChannel <- podStatusSyncRequest{pod.UID, *newStatus}:
```

Comment:
We can do what SetPodStatus does, although now that we have the periodic sync, I actually wonder which approach is better. Perhaps it would be better to drop the update in the full channel case in both places and let the periodic sync pick up the update? Perhaps I'm over optimizing here.